R/pkg/R/functions.R

```diff
@@ -2836,7 +2845,11 @@ setMethod("lpad", signature(x = "Column", len = "numeric", pad = "character"),
 setMethod("rand", signature(seed = "missing"),
           function(seed) {
             jc <- callJStatic("org.apache.spark.sql.functions", "rand")
-            column(jc)
+
+            # By assigning a one-row data.frame, the result of this function can be collected
+            # returning a one-element Column
+            df <- as.DataFrame(sparkRSQL.init(), data.frame(0))
```

Comment:
See my comment from March 30 to illustrate why this is needed. I'll change sparkRSQL.init() to sparkR.session(). Thanks for catching this!